ml-agents/mlagents/trainers/trainer_controller.py

```diff
@@ -72,6 +83,8 @@ def __init__(
         self.fast_simulation = fast_simulation
         np.random.seed(self.seed)
         tf.set_random_seed(self.seed)
+        self.sampler_manager = SamplerManager(reset_param_dict) if reset_param_dict is not None else None
```

Comment:
This feels a little inconsistent - you create the SampleManager in here from the config, but the LessonController is created externally and passed. I think passing both as objects (instead) is slightly better.

Reply:
It might make the code cleaner if you always create a SampleManager. A SamplerManager with an empty dictionary of reset parameters should be valid (it just won't return any reset parameters). Not sure how that interacts with the lesson_controller logic though.